src/system/PageLockModule/Api/LockingApi.php

```diff
@@ -0,0 +1,262 @@
+<?php
+/**
+ * This file is part of the Zikula package.
+ *
+ * Copyright Zikula Foundation - http://zikula.org/
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zikula\PageLockModule\Api;
+
+use PageUtil;
+use Twig_Environment;
+use Zikula\PageLockModule\Entity\PageLockEntity;
+
+/**
+ * Class LockingApi.
+ * @package Zikula\PageLockModule\Api
+ *
+ * This class provides means for using a locking mechanism.
+ * It should be used instead of the old user api.
+ */
+class LockingApi
+{
+    /**
+     * length of time to lock a page
+     */
+    const PAGELOCKLIFETIME = 30;
+
+    /**
+     * @var Twig_Environment
+     */
+    private $twig;
+
+    /**
+     * LockingApi constructor.
+     *
+     * @param Twig_Environment $twig Twig service instance.
+     */
+    public function __construct(Twig_Environment $twig)
+    {
+        $this->twig = $twig;
+    }
+
+    /**
+     * Add the page locking code to the page header
+     *
+     * @param string $lockName        The name of the lock to be released
+     * @param string $returnUrl       The URL to return control to (optional) (default: null)
+     * @param bool   $ignoreEmptyLock Ignore an empty lock name (optional) (default: false)
+     *
+     * @return bool true
+     */
+    public function pageLock($lockName, $returnUrl = null, $ignoreEmptyLock = false)
+    {
+        $hasLock = true;
+        $lockedHtml = '';
+
+        if (!empty($lockName) || !$ignoreEmptyLock) {
+            PageUtil::addVar('javascript', 'zikula.ui');
+            PageUtil::addVar('javascript', 'system/PageLockModule/Resources/public/js/PageLock.js');
+            PageUtil::addVar('stylesheet', \ThemeUtil::getModuleStylesheet('ZikulaPageLockModule'));
+
+            $lockInfo = $this->requireLock($lockName, \UserUtil::getVar('uname'), $_SERVER['REMOTE_ADDR']);
```

Comment:
also need proper replacement for `$_SERVER['REMOTE_ADDR']` either from Request (request_stack) or something else.